tests/test_pipelines.py

```diff
@@ -1214,6 +1214,37 @@ def test_stable_diffusion_memory_chunking(self):
         assert mem_bytes > 3.75 * 10**9
         assert np.abs(image_chunked.flatten() - image.flatten()).max() < 1e-3
 
+    @slow
+    @unittest.skipIf(torch_device == "cpu", "Stable diffusion is supposed to run on GPU")
+    def test_stable_diffusion_text2img_pipeline_fp16(self):
+        torch.cuda.reset_peak_memory_stats()
+        model_id = "CompVis/stable-diffusion-v1-4"
+        pipe = StableDiffusionPipeline.from_pretrained(
+            model_id, revision="fp16", torch_dtype=torch.float16, use_auth_token=True
+        ).to(torch_device)
+        pipe.set_progress_bar_config(disable=None)
+
+        prompt = "a photograph of an astronaut riding a horse"
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        output_chunked = pipe(
+            [prompt], generator=generator, guidance_scale=7.5, num_inference_steps=10, output_type="numpy"
+        )
+        image_chunked = output_chunked.images
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        with torch.autocast(torch_device):
```

Comment:
@Narsil I dont understand why we're applying autocast to `pipe`if it's already in fp16? We probably want to cast it to fp32 before this?

Reply:
Because `autocast` will use both `fp32` and `fp16` operations, regardless of weights/input tensors.

The resulting images will be different, and this test tries to prove that the difference is minimal (2% total variance).
This is important to keep in mind, because **maybe** some operations should be kept in fp32 because the quality of the output would be necessary.
Operations like softmax for instance.